modules/post/windows/manage/ducky.rb

```diff
@@ -0,0 +1,137 @@
+                  '107' => 'ADD', '109' => 'SUBTRACT', '110' => 'DECIMAL', '111' => 'DIVIDE', '112' => 'F1',
+                  '113' => 'F2', '114' => 'F3', '115' => 'F4', '116' => 'F5', '117' => 'F6',
+                  '118' => 'F7', '119' => 'F8', '120' => 'F9', '121' => 'F10', '122' => 'F11',
+                  '123' => 'F12','188' => 'COMMA','190' => 'PERIOD','191' => 'SLASH','48' => '0',
+                  '49' => '1','50' => '2','51' => '3','52' => '4','53' => '5',
+                  '54' => '6', '55' => '7', '56' => '8', '57' => '9'
+                }
+    # Define keypress actions
+    actions = { '0' => 'press', '1' => 'down', '2' => 'up' }
+    # Parse DuckyScript comments
+    if line.starts_with?('REM ')
+      return
+    end
+    # Cursor movement 
+    # Althought this is a bit hacky
+    cursor_line = line.strip
+    if ( cursor_line.eql?('UP') || cursor_line.eql?('DOWN') || cursor_line.eql?('LEFT') || cursor_line.eql?('RIGHT') || cursor_line.eql?('BACKSPACE') ||
+      cursor_line.eql?('TAB') || cursor_line.eql?('PAGEUP') || cursor_line.eql?('PAGEDOWN') || cursor_line.eql?('HOME') ||
+      cursor_line.eql?('END') || cursor_line.eql?('SPACE') )
+      session.ui.keyevent_send(key_codes.key(cursor_line).to_i, actions.key('press').to_i)
+    end
+    if line.include?('STRING') || line.include?('STRINGLN') || line.include?('GUI')
+      line_array = line.split(' ', 2)
+      # Write a string and press ENTER
+      if line_array[0] == 'STRINGLN'
+        session.ui.keyboard_send(line_array[1])
+        session.ui.keyevent_send(key_codes.key('RETURN').to_i, actions.key('press').to_i)
+      end
+      # Write a string
+      if line_array[0] == 'STRING'
+        session.ui.keyboard_send(line_array[1])
+      end
+      # Press Windows + R to launch run dialog
+      if line_array[0] == 'GUI'
+        session.ui.keyevent_send(key_codes.key('WINLEFT').to_i, actions.key('down').to_i)
+        session.ui.keyevent_send(key_codes.key('R').to_i, actions.key('press').to_i)
+        session.ui.keyevent_send(key_codes.key('WINLEFT').to_i, actions.key('up').to_i)
+      end
+    end
+    if line.include?('KEYEVENT')
+      line_array = line.split(' ')
+      # Parse raw keyevents; example format, KEYEVENT 82 press
+      if line_array[0] == 'KEYEVENT'
+        session.ui.keyevent_send(line_array[1].to_i, actions.key(line_array[2]).to_i)
+      end
+    end
+    # Parse F Keys
+    if line.starts_with?('F')
+      session.ui.keyevent_send(key_codes.key(line).to_i, actions.key('press').to_i)
+    end
+  end
+
+  def run
+    return 0 if session.type != 'meterpreter'
+    if datastore['FILENAME'].blank?
+      print_error('A file needs to be provided!')
+      return 0
+    end
+    print_good("Reading file #{datastore['FILENAME']}")
+    File.readlines(datastore['FILENAME']).each do |line|
+      # Parse out blank lines
+      line.split(/\r?\n/)
```

Comment:
I don't think this line does anything

Reply:
Sure enough! It doesn't; it was an artifact from some initial testing. I have removed it and will commit updated changes ASAP.

Reply:
This has been removed from the latest push.